instrumentation-tools/src/api_surface.py

```diff
@@ -0,0 +1,60 @@
+#  Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+#  This product includes software developed at Datadog (https://www.datadoghq.com/).
+#  Copyright 2016-Present Datadog, Inc.
+
+import re
+from math import floor
+
+DIVIDER = "#"
+TESTABLE_API_REG_EX = re.compile("^(.+)(constructor|fun)(.+)")
+ROOT_REG_EX = re.compile("^(?<!companion)(\\s*)[a-zA-z]*(class|interface|enum|object)(.*)(\\.|\\s)(.+)$")
+
+
+def sanitize_prefix(previous_indent: int,
```

Comment:
Doesn't `(\\s*)` in `ROOT_REG_EX` already capture the indentation? Anyway, I feel that we can get rid of the entire `sanitize_prefix` by just utilising this regexp somehow more, no 🙂?

Reply:
No, it's actually more complicated than that. Let's say you have an api Surface that is:
```
class A
    fun a
    fun b
class B
    fun d
    fun e
 class C
    fun f
    fun g
class D
    fun h
```
then your test cases IDs should look : `A#fun a`,...,`B#fun d`, `B#C#fun f` ..., `D#fun h`. Based on the indentation level on your group `class A`, `class B`, ` class C` you will have to remove a group or more from the previous prefix -> 
`B#C` will become `D` at next line.

Reply:
I see now 👌  - it's way cleaner with the docstring 👍